common/lib/opentelemetry/common/http/request_attributes.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Common
+    module HTTP
+      # RequestAttributes contains common helpers for setting http client attributes in spans
+      module RequestAttributes
+        extend self
+
+        def from_request(request_method, uri, config = {})
+          uri = hide_query_params(uri) if config[:hide_query_params]
```

Comment:
I think there are a lot of ways to go with this implementation under the hood but overall I like this config option and think it's a reasonable thing to add.  Generally speaking though, for http client helpers, it's an interesting dilemma.

On one hand, you could add more config for more use cases with a middle ground approach here to add some or all query parameters as a span attribute. `hide_query_params` could optionally take an allow-llst maybe, or if changed to `show_query_params` could look something like:

```
require `cgi`
def show_query_params(uri, span)
  if config[:show_query_params] && config[:show_query_params].length > 0

    return if config[:show_query_params][0] == 'all'

    params = CGI::parse(uri.query)

    config[:show_query_params].each do |query_param|
      if params[query_param]
         span.set_attribute("request.#{query_param}", params[query_param])
      end
    end
  else
    uri.query = '' if uri.query
    uri
  end
end
```

But, maybe the ideal end state is to allow users an arbitrary proc on the span. `from_request` could have a similar method like `after_request` and users can do whatever they want to their spans.

```
def after_request(span, request_method, uri, config = {})
  if config[:after_request]
    after_request = config[:after_request]
    after_request.call(span, request_method,uri,config)
  end

  span
end
```

Anyway, just something to think about for future work but this is a great improvement so it'll be great to see this merged and available.